hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/OmSnapshotUtils.java

```diff
@@ -128,7 +130,12 @@ public static void createHardLinks(Path dbPath) throws IOException {
                   "Failed to create directory: " + parent.toString());
             }
           }
-          Files.createLink(fullToPath, fullFromPath);
+
+          if (isSamePartition(fullFromPath.getParent(), fullToPath.getParent())) {
+            Files.createLink(fullToPath, fullFromPath);
+          } else {
+            Files.copy(fullFromPath, fullToPath, StandardCopyOption.REPLACE_EXISTING);
+          }
```

Comment:
nit:
1. this code can be moved to helper function and reuse here and in line 193-196.
2. Should we change `createHardLinks` function name as well? Since it is just not creating hard links.